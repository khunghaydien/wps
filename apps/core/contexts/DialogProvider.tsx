import * as React from 'react';

import DialogRoot from '../blocks/Dialog/DialogRoot';
import DialogContext, { DialogType, DialogTypeProps } from './DialogContext';

type Props = {
  children: React.ReactNode;
};

const DialogProvider = ({ children }: Props) => {
  const [dialogs, setDialogs] = React.useState({});
  const showDialog = React.useCallback(
    (key: string, dialog: DialogType, props: DialogTypeProps) => {
      setDialogs((ds) => ({
        ...ds,
        [key]: { dialog, props },
      }));
    },
    []
  );
  const hideDialog = React.useCallback((key: string) => {
    setDialogs((ds) => {
      if (ds[key]) {
        const newDialogs = { ...ds };
        delete newDialogs[key];
        return newDialogs;
      } else {
        return ds;
      }
    });
  }, []);

  return (
    <DialogContext.Provider
      value={{
        showDialog,
        hideDialog,
      }}
    >
      <>
        {children}
        <DialogRoot dialogs={dialogs} />
      </>
    </DialogContext.Provider>
  );
};

/**
 * A Provide for modal or modeless dialogs.
 *
 * @example
 * // A sample of Dialog component
 * const JobSelectDialog = <Dialog
 *   modal
 *   id='unique-key' // must be unique since it is used for identifying a dialog.
 *   content={...}
 *   footer={...}
 * />;
 *
 * // Wrap DialogProvider around app root
 * <DialogProvider>
 *  <App/>
 * </DialogProvider>
 *
 * // The client code can show or hide a dialog using useDialog API.
 * const [ showJobDialog, hideJobDialog ] = useDialog(JobSelectDialog); // Specify a dialog you want
 * return <div><button onClick={showDialog} /></div>;
 */
export default React.memo<Props>(DialogProvider);
