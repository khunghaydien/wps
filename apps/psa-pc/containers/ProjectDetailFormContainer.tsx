import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { AnyAction, bindActionCreators } from 'redux';
import { ThunkDispatch } from 'redux-thunk';

import { Formik } from 'formik';
import cloneDeep from 'lodash/cloneDeep';

import projectDetailFormSchema from '@psa/schema/ProjectDetailForm';

import { catchBusinessError, confirm } from '@apps/commons/actions/app';
import msg from '@apps/commons/languages';

import {
  searchClientList as searchClientListApi,
  searchEmployeeList as searchEmployeeListApi,
} from '@apps/domain/models/psa/Project';

import { State } from '@psa/modules';
import { actions as modeActions } from '@psa/modules/ui/mode';
import { SIDEBAR_TYPES } from '@psa/modules/ui/sidebar';

import {
  backToProjectList,
  saveProject,
} from '@psa/action-dispatchers/Project';
import saveProjectManagerLocally from '@psa/action-dispatchers/ProjectManager';
import {
  openDeleteProjectDialog,
  switchToSidebar,
} from '@psa/action-dispatchers/PSA';
import { getExtendedItemList } from '@psa/action-dispatchers/PsaExtendedItem';
import getResourcegroups from '@psa/action-dispatchers/ResourceGroup';
import { getGroupMembers } from '@resource/action-dispatchers/ResourceGroup';

import ProjectDetail from '@psa/components/ProjectScreen/ProjectDetail';

// Selector
export function getManagerList(suggestList: any) {
  const { employeeList, records } = suggestList;
  const results = employeeList || records;

  return results && results.length > 0
    ? results.map((employee) => ({
        code: employee.code,
        id: employee.id,
        name: employee.displayName,
      }))
    : results;
}

export function getCalendarList(state: any) {
  const { calendarList } = state.entities;

  return calendarList && calendarList.length > 0
    ? calendarList.map((calendar) => ({
        value: calendar.id,
        text: calendar.name,
      }))
    : calendarList;
}

const ProjectDetailFormContainer = () => {
  const dispatch = useDispatch() as ThunkDispatch<State, void, AnyAction>;

  const activeDialog = useSelector(
    (state: State) => state.ui.dialog.activeDialog
  );
  const activityList = useSelector(
    (state: State) => state.entities.psa.activity.activityList
  );
  const calendarListOption = useSelector((state: State) =>
    getCalendarList(state)
  );
  const companyId = useSelector((state: State) => state.userSetting.companyId);
  const deptSuggestList = useSelector(
    (state: State) => state.entities.departmentList
  );
  const employeeId = useSelector(
    (state: State) => state.userSetting.employeeId
  );
  const extendedItemConfigList = useSelector(
    (state: State) => state.entities.psa.psaExtendedItem.project
  );
  const groupDetail = useSelector(
    (state: State) => state.entities.psa.resourceGroup.detail
  );
  const managerListOption = useSelector((state: State) =>
    getManagerList(state.entities)
  );
  const mode = useSelector((state: State) => state.ui.mode);
  const pageNum = useSelector(
    (state: State) => state.entities.psa.project.pageNum
  );
  const projectListFilterState = useSelector(
    (state: State) => state.ui.filter.project
  );
  const projectManagerId = useSelector(
    (state: State) => state.entities.psa.projectManager.projectManagerId
  );
  const resourceGroupList = useSelector(
    (state: State) => state.entities.psa.resourceGroup.groups
  );
  const resourceManagerList = useSelector(
    (state: State) => state.entities.psa.resourceManager.resourceManagerList
  );
  const selectedProject = useSelector(
    (state: State) => state.entities.psa.project.project
  );
  const useExistingJobCode = useSelector(
    (state: State) => state.entities.psa.setting.useExistingJobCode
  );
  const selectedGroup = useSelector(
    (state: State) => state.entities.psa.psaGroup.selectedGroup
  );

  useEffect(() => {
    dispatch(getResourcegroups(companyId, selectedGroup.id));
    dispatch(getExtendedItemList(companyId, 'Project'));
  }, []);

  const onClickCancel = () => {
    dispatch(
      confirm(msg().Psa_Msg_ConfirmDiscardChanges, (yes) => {
        if (yes) {
          dispatch(switchToSidebar(SIDEBAR_TYPES.ACTIVITY));
        }
      })
    );
  };

  const onClickBackToProjectList = () => {
    dispatch(
      backToProjectList(
        companyId,
        pageNum,
        selectedGroup.id,
        projectListFilterState
      )
    );
  };

  const searchEmployeeList = (value) =>
    searchEmployeeListApi(companyId, value, selectedGroup.id);

  const generateInitialValues = () => {
    // Transform the object such that id to serve as unique identifier for formik validation
    const transformedExtendeditems = {};
    selectedProject.extendedItems &&
      selectedProject.extendedItems.length > 0 &&
      selectedProject.extendedItems.forEach((x) => {
        transformedExtendeditems[x.id] = x.value;
      });
    return {
      code: selectedProject.code,
      companyId: selectedProject.companyId,
      description: selectedProject.description,
      clientId: selectedProject.clientId,
      deptId: selectedProject.deptId,
      endDate: selectedProject.endDate,
      pmBaseId: selectedProject.pmBaseId,
      groupId: selectedProject.groupId,
      name: selectedProject.name,
      projectId: selectedProject.projectId,
      startDate: selectedProject.startDate,
      status: selectedProject.status,
      jobId: selectedProject.jobId || '',
      jobCode: selectedProject.jobCode || '',
      workTimePerDay: selectedProject.workTimePerDay || '',
      calendarId: selectedProject.calendarId,
      workingDaySUN: selectedProject.workingDaySUN,
      workingDayMON: selectedProject.workingDayMON,
      workingDayTUE: selectedProject.workingDayTUE,
      workingDayWED: selectedProject.workingDayWED,
      workingDayTHU: selectedProject.workingDayTHU,
      workingDayFRI: selectedProject.workingDayFRI,
      workingDaySAT: selectedProject.workingDaySAT,
      ...transformedExtendeditems,
      extendedItemConfigList: extendedItemConfigList || [],
      useExistingJobCode,
      psaGroupId: selectedGroup.id,
    };
  };

  const handleSubmit = (values) => {
    const project: any = cloneDeep(values);
    const extendedItems = [];
    if (extendedItemConfigList) {
      extendedItemConfigList.forEach((eI) => {
        extendedItems.push({
          id: eI.id,
          value: values[eI.id],
          name: eI.name,
        });
        delete project[eI.id];
      });
    }
    delete project.extendedItemConfigList;
    project.extendedItems = extendedItems;
    selectedProject.extendedItemValueId &&
    selectedProject.extendedItemValueId !== ''
      ? (project.extendedItemValueId = selectedProject.extendedItemValueId)
      : (project.extendedItemValueId = '');

    // remove ui values
    if (!useExistingJobCode) {
      delete project.jobId;
    }
    delete project.jobCode;
    delete project.useExistingJobCode;

    dispatch(saveProject(project, companyId, selectedGroup.id));
  };

  const Actions = bindActionCreators(
    {
      catchBusinessError,
      editProjectMode: modeActions.editProject,
      onClickDeleteProject: openDeleteProjectDialog,
      saveProjectManagerLocally,
      getGroupMembers,
    },
    dispatch
  );

  return (
    <Formik
      enableReinitialize
      initialValues={generateInitialValues()}
      validationSchema={projectDetailFormSchema}
      onSubmit={handleSubmit}
    >
      {(props) => {
        return (
          <ProjectDetail
            activeDialog={activeDialog}
            activityList={activityList}
            backToProjectList={onClickBackToProjectList}
            calendarListOption={calendarListOption}
            companyId={companyId}
            deptSuggestList={deptSuggestList}
            dirty={props.dirty}
            editProjectMode={Actions.editProjectMode}
            employeeId={employeeId}
            errors={props.errors}
            extendedItemConfigList={extendedItemConfigList}
            getGroupMembers={Actions.getGroupMembers}
            getManagerList={getManagerList}
            groupDetail={groupDetail}
            handleSubmit={props.handleSubmit}
            managerListOption={managerListOption}
            mode={mode}
            onClickCancel={onClickCancel}
            onClickDeleteProject={Actions.onClickDeleteProject}
            projectManagerId={projectManagerId}
            resourceGroupList={resourceGroupList}
            resourceManagerList={resourceManagerList}
            saveProjectManagerLocally={Actions.saveProjectManagerLocally}
            searchEmployeeList={searchEmployeeList}
            searchClientList={searchClientListApi}
            selectedProject={selectedProject}
            setFieldValue={props.setFieldValue}
            touched={props.touched}
            useExistingJobCode={useExistingJobCode}
            values={props.values}
            catchBusinessError={Actions.catchBusinessError}
            selectedGroupId={selectedGroup.id}
          />
        );
      }}
    </Formik>
  );
};

export default ProjectDetailFormContainer;
