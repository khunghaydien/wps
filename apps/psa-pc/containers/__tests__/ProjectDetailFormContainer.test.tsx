import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import DialogProvider from '@apps/core/contexts/DialogProvider';

import ProjectDetailFormContainer from '../ProjectDetailFormContainer';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
describe('ProjectDetailFormContainer Test', () => {
  let store;
  let component;

  const initialState = {
    entities: {
      departmentList: [],
      psa: {
        project: {
          project: {},
          pageSize: 1,
        },
        activity: {
          activityList: [],
        },
        psaExtendedItem: {},
        resourceGroup: {
          detail: [],
          groups: [],
        },
        psaGroup: {
          selectedGroup: {},
        },
        projectManager: {},
        resourceManager: {},
        setting: {},
        resource: {},
        request: {},
        role: {
          role: {},
        },
        // projectFinance:{}
      },
      clientInfo: {
        opportunity: {},
        client: {},
      },
      employeeList: [],
    },
    ui: {
      filter: {
        project: {},
      },
      mode: '',
      dialog: {
        activeDialog: [],
      },
    },
    userSetting: {
      companyId: '',
    },
  };
  beforeEach(() => {
    store = mockStore(initialState);
    component = renderer.create(
      <Provider store={store}>
        <DialogProvider>
          <ProjectDetailFormContainer />
        </DialogProvider>
      </Provider>
    );
  });
  it('it renders without crashing', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
