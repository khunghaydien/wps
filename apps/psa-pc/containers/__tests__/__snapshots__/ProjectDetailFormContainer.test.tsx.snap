// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectDetailFormContainer Test it renders without crashing 1`] = `
<div
  className="ts-psa__project"
>
  <div
    className="ts-psa__common-header"
  >
    <div
      className="ts-psa__common-header-left"
    >
      <span
        className="ts-psa__common-header-breadcrumb"
      />
      <span
        className="ts-psa__common-header-title"
      >
        Project Details
      </span>
    </div>
    <div
      className="ts-psa__common-header__btn-area"
    >
      <button
        className="ts-psa__project__btn--delete ts-button ts-button--default"
        data-testid="ts-psa__project__btn--delete"
        disabled={false}
        onClick={[Function]}
        type="button"
        value=""
      >
        <div
          className="ts-button__contents"
        >
          <span
            className="ts-button__text"
          >
            Delete Project
          </span>
        </div>
      </button>
      <button
        className="ts-psa__project__btn--cancel ts-button ts-button--default"
        data-test-id="ts-psa__project__btn--cancel"
        disabled={true}
        onClick={[Function]}
        type="button"
        value=""
      >
        <div
          className="ts-button__contents"
        >
          <span
            className="ts-button__text"
          >
            Cancel
          </span>
        </div>
      </button>
      <button
        className="ts-psa__project__btn--save ts-button ts-button--primary"
        data-test-id="ts-psa__project__btn--save"
        disabled={true}
        onClick={[Function]}
        type="button"
        value=""
      >
        <div
          className="ts-button__contents"
        >
          <span
            className="ts-button__text"
          >
            Save
          </span>
        </div>
      </button>
    </div>
  </div>
  <div
    className="ts-psa__project__content"
  >
    <div
      className="ts-psa__project-form"
    >
      <section
        className="ts-psa__project-form__base"
      >
        <h3
          className="ts-psa__project-form__title"
        >
          Base Info
        </h3>
        <div
          className="ts-psa__form-field  "
          data-testid="ts-psa__project-form__base__code"
        >
          <div
            className="ts-psa__form-field__label"
          >
            <span
              className="is-required"
            >
              *
            </span>
             
            {$Psa_Clbl_ProjectCode}
          </div>
          <input
            className="ts-text-field slds-input"
            disabled={true}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder={null}
            readOnly={false}
            type="text"
            value=""
          />
        </div>
        <div
          className="ts-psa__form-field  "
          data-testid="ts-psa__project-form__base__name"
        >
          <div
            className="ts-psa__form-field__label"
          >
            <span
              className="is-required"
            >
              *
            </span>
             
            Project Title
          </div>
          <input
            className="ts-text-field slds-input"
            disabled={true}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder={null}
            readOnly={false}
            type="text"
            value=""
          />
        </div>
        <div
          className="ts-psa__form-field ts-psa__project-form__base__desc "
          data-testid="ts-psa__project-form__base__description"
        >
          <div
            className="ts-psa__form-field__label"
          >
             
            Description
          </div>
          <textarea
            className="ts-textarea-field slds-input"
            disabled={true}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            value=""
          />
        </div>
        <div
          className="ts-psa__form-field  "
          data-testid="ts-psa__project-form__base__pm"
        >
          <div
            className="ts-psa__form-field__label"
          >
            <span
              className="is-required"
            >
              *
            </span>
             
            Project Manager
          </div>
          <div
            aria-expanded={false}
            aria-haspopup="listbox"
            aria-owns="react-autowhatever-1"
            className="react-autosuggest__container"
            role="combobox"
          >
            <div>
              <input
                aria-activedescendant={null}
                aria-autocomplete="list"
                aria-controls="react-autowhatever-1"
                autoComplete="off"
                className="ts-text-field slds-input"
                disabled={true}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                readOnly={false}
                type="text"
                value="undefined"
              />
            </div>
            <div
              className="react-autosuggest__suggestions-container"
              id="react-autowhatever-1"
              role="listbox"
            />
          </div>
        </div>
        <div
          className="ts-psa__form-field  "
          data-testid="ts-psa__project-form__base__project-client"
        >
          <div
            className="ts-psa__form-field__label"
          >
             
            Client
            <div
              className="commons-tooltip "
              id="commons-tooltip"
            >
              <div
                className="slds-tooltip-trigger"
                style={
                  Object {
                    "display": "block",
                  }
                }
              >
                <test-file-stub
                  aria-describedby="ts-psa__form-field__tooltip"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={Object {}}
                  tabIndex={0}
                />
                <span />
              </div>
            </div>
          </div>
          <div
            aria-expanded={false}
            aria-haspopup="listbox"
            aria-owns="react-autowhatever-1"
            className="react-autosuggest__container"
            role="combobox"
          >
            <div>
              <input
                aria-activedescendant={null}
                aria-autocomplete="list"
                aria-controls="react-autowhatever-1"
                autoComplete="off"
                className="ts-text-field slds-input"
                disabled={true}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                placeholder="Search Project Client"
                readOnly={false}
                type="text"
                value=""
              />
            </div>
            <div
              className="react-autosuggest__suggestions-container"
              id="react-autowhatever-1"
              role="listbox"
            />
          </div>
        </div>
        <div
          className="ts-psa__form-field  "
          data-testid="ts-psa__project-form__base__project-department"
        >
          <div
            className="ts-psa__form-field__label"
          >
             
            Project Department
          </div>
          <div
            aria-expanded={false}
            aria-haspopup="listbox"
            aria-owns="react-autowhatever-1"
            className="react-autosuggest__container"
            role="combobox"
          >
            <div>
              <input
                aria-activedescendant={null}
                aria-autocomplete="list"
                aria-controls="react-autowhatever-1"
                autoComplete="off"
                className="ts-text-field slds-input"
                disabled={true}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                placeholder="Search Project Department"
                readOnly={false}
                type="text"
                value=""
              />
            </div>
            <div
              className="react-autosuggest__suggestions-container"
              id="react-autowhatever-1"
              role="listbox"
            />
          </div>
        </div>
        <div
          className="ts-psa__form-field  "
          data-testid="ts-psa__project-form__base__rg"
        >
          <div
            className="ts-psa__form-field__label"
          >
             
            Resource Group
          </div>
          <div
            className="ts-Resource-group-filter"
          >
            <div
              className="ts-Resource-group-filter__selection"
            >
              <div
                className="ts-Resource-group-filter__selection-button"
              >
                <test-file-stub
                  className="ts-Resource-group-filter__selection-button__icon__action"
                />
                <div
                  onClick={[Function]}
                >
                  <span
                    className="ts-Resource-group-filter__selection-button__name__action"
                  >
                    Select Resource Group
                  </span>
                </div>
              </div>
              <div
                className="ts-Resource-group-filter__selection-list"
              />
            </div>
          </div>
        </div>
        <div
          className="ts-psa__form-field  "
          data-testid="ts-psa__project-form__base__work-hours"
        >
          <div
            className="ts-psa__form-field__label"
          >
            <span
              className="is-required"
            >
              *
            </span>
             
            Work Hours Per Day
          </div>
          <input
            className="ts-text-field slds-input"
            disabled={true}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder={null}
            readOnly={false}
            type="tel"
            value="0"
          />
        </div>
        <div
          className="ts-psa__form-field ts-psa__project-form__base__date "
        >
          <div
            className="ts-psa__form-field__label"
          >
             
            Project Duration
          </div>
          <div
            className="ts-psa__date-field__container"
          >
            <div
              className="commons-fields-date-field"
              data-testid=""
            >
              <div>
                <div
                  className="react-datepicker-wrapper"
                >
                  <div
                    className="react-datepicker__input-container"
                  >
                    <input
                      className="slds-input"
                      disabled={true}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      placeholder="Start Date"
                      readOnly={false}
                      required={false}
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              className="ts-psa__date-field__calendar-icon-wrapper"
            >
              <test-file-stub
                className="ts-psa__date-field__calendar-icon-body"
              />
            </div>
          </div>
          <span
            className="separator"
          >
            -
          </span>
          <div
            className="ts-psa__date-field__container"
          >
            <div
              className="commons-fields-date-field"
              data-testid=""
            >
              <div>
                <div
                  className="react-datepicker-wrapper"
                >
                  <div
                    className="react-datepicker__input-container"
                  >
                    <input
                      className="slds-input"
                      disabled={true}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      placeholder="End Date"
                      readOnly={false}
                      required={false}
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              className="ts-psa__date-field__calendar-icon-wrapper"
            >
              <test-file-stub
                className="ts-psa__date-field__calendar-icon-body"
              />
            </div>
          </div>
        </div>
        <div
          className="ts-psa__form-field  "
          data-testid="ts-psa__project-form__base__status"
        >
          <div
            className="ts-psa__form-field__label"
          >
             
            Status
          </div>
          <select
            className="slds-select ts-select"
            disabled={false}
            id=""
            multiple={false}
            onChange={[Function]}
            required={false}
            size={0}
            value=""
          >
            <option
              disabled={null}
              value="Planning"
            >
              Planning
            </option>
            <option
              disabled={null}
              value="InProgress"
            >
              In Progress
            </option>
            <option
              disabled={null}
              value="Completed"
            >
              Completed
            </option>
            <option
              disabled={null}
              value="Cancelled"
            >
              Cancelled
            </option>
          </select>
        </div>
        <div
          className="ts-psa__form-field ts-psa__project-form__base__calendar "
          data-testid="ts-psa__project-form__base__calendar"
        >
          <div
            className="ts-psa__form-field__label"
          >
             
            Calendar
          </div>
          <select
            className="slds-select ts-select ts-psa__project-form__base__calendar-contents"
            disabled={true}
            id=""
            multiple={false}
            onChange={[Function]}
            required={false}
            size={0}
            value=""
          />
        </div>
        <div
          className="ts-psa__form-field ts-psa__project-form__base__workingdays "
          data-testid="ts-psa__project-form__base__working-days"
        >
          <div
            className="ts-psa__form-field__label"
          >
             
            Project Working Days
          </div>
          <div
            className="ts-psa__project-form__base__workingdays-contents"
          >
            <label
              className="ts-psa__project-form__base__workingdays-label"
              htmlFor="workingDaySUN"
            >
              <input
                checked={false}
                className="ts-psa__project-form__base__workingdays-checkbox"
                disabled={true}
                id="workingDaySUN"
                onChange={[Function]}
                type="checkbox"
                value={false}
              />
              <span
                className="ts-psa__project-form__base__workingdays-text"
              >
                Sun
              </span>
            </label>
            <label
              className="ts-psa__project-form__base__workingdays-label"
              htmlFor="workingDayMON"
            >
              <input
                checked={false}
                className="ts-psa__project-form__base__workingdays-checkbox"
                disabled={true}
                id="workingDayMON"
                onChange={[Function]}
                type="checkbox"
                value={false}
              />
              <span
                className="ts-psa__project-form__base__workingdays-text"
              >
                Mon
              </span>
            </label>
            <label
              className="ts-psa__project-form__base__workingdays-label"
              htmlFor="workingDayTUE"
            >
              <input
                checked={false}
                className="ts-psa__project-form__base__workingdays-checkbox"
                disabled={true}
                id="workingDayTUE"
                onChange={[Function]}
                type="checkbox"
                value={false}
              />
              <span
                className="ts-psa__project-form__base__workingdays-text"
              >
                Tue
              </span>
            </label>
            <label
              className="ts-psa__project-form__base__workingdays-label"
              htmlFor="workingDayWED"
            >
              <input
                checked={false}
                className="ts-psa__project-form__base__workingdays-checkbox"
                disabled={true}
                id="workingDayWED"
                onChange={[Function]}
                type="checkbox"
                value={false}
              />
              <span
                className="ts-psa__project-form__base__workingdays-text"
              >
                Wed
              </span>
            </label>
            <label
              className="ts-psa__project-form__base__workingdays-label"
              htmlFor="workingDayTHU"
            >
              <input
                checked={false}
                className="ts-psa__project-form__base__workingdays-checkbox"
                disabled={true}
                id="workingDayTHU"
                onChange={[Function]}
                type="checkbox"
                value={false}
              />
              <span
                className="ts-psa__project-form__base__workingdays-text"
              >
                Thu
              </span>
            </label>
            <label
              className="ts-psa__project-form__base__workingdays-label"
              htmlFor="workingDayFRI"
            >
              <input
                checked={false}
                className="ts-psa__project-form__base__workingdays-checkbox"
                disabled={true}
                id="workingDayFRI"
                onChange={[Function]}
                type="checkbox"
                value={false}
              />
              <span
                className="ts-psa__project-form__base__workingdays-text"
              >
                Fri
              </span>
            </label>
            <label
              className="ts-psa__project-form__base__workingdays-label"
              htmlFor="workingDaySAT"
            >
              <input
                checked={false}
                className="ts-psa__project-form__base__workingdays-checkbox"
                disabled={true}
                id="workingDaySAT"
                onChange={[Function]}
                type="checkbox"
                value={false}
              />
              <span
                className="ts-psa__project-form__base__workingdays-text"
              >
                Sat
              </span>
            </label>
          </div>
        </div>
        <div />
      </section>
      <section
        className="ts-psa__project-form__client"
      >
        <h3
          className="ts-psa__project-form__title"
        >
          Client Info
        </h3>
        <div
          className="ts-psa__form-field  "
        >
          <div
            className="ts-psa__form-field__label"
          >
             
            Client (Salesforce Account)
          </div>
          <input
            className="ts-text-field slds-input"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder={null}
            readOnly={false}
            type="text"
            value=""
          />
        </div>
        <div
          className="ts-psa__form-field  "
        >
          <div
            className="ts-psa__form-field__label"
          >
             
            Opportunity (Salesforce Account)
          </div>
          <input
            className="ts-text-field slds-input"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder={null}
            readOnly={false}
            type="text"
            value=""
          />
        </div>
        <div
          className="ts-psa__form-field  "
        >
          <div
            className="ts-psa__form-field__label"
          >
             
            Client Person in Charge (Salesforce Contact)
          </div>
          <input
            className="ts-text-field slds-input"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder={null}
            readOnly={false}
            type="text"
            value=""
          />
        </div>
        <div
          className="ts-psa__form-field  "
        >
          <div
            className="ts-psa__form-field__label"
          >
             
            Contract Type
          </div>
          <select
            className="slds-select ts-select"
            disabled={false}
            id=""
            multiple={false}
            onChange={[Function]}
            required={false}
            size={0}
            value=""
          >
            <option
              disabled={null}
              value="flex"
            >
              Flex
            </option>
            <option
              disabled={null}
              value="permanent"
            >
              Permanent
            </option>
          </select>
        </div>
      </section>
      <section
        className="ts-psa__project-form__finance"
      >
        <h3
          className="ts-psa__project-form__title"
        >
          Financial Info
        </h3>
        <p
          className="ts-psa__project-form__finance__subtitle"
        >
          Target Revenue
        </p>
        <div
          className="ts-psa__form-field  "
        >
          <div
            className="ts-psa__form-field__label"
          >
             
            Target Revenue
          </div>
          <input
            className="ts-text-field slds-input"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder={null}
            readOnly={false}
            type="text"
            value=""
          />
        </div>
        <div
          className="ts-psa__form-field  "
        >
          <div
            className="ts-psa__form-field__label"
          >
             
            Target Margin
          </div>
          <input
            className="ts-text-field slds-input"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder={null}
            readOnly={false}
            type="text"
            value=""
          />
        </div>
        <div
          className="ts-psa__form-field  "
        >
          <div
            className="ts-psa__form-field__label"
          >
             
            Target Cost
          </div>
          <input
            className="ts-text-field slds-input"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder={null}
            readOnly={false}
            type="text"
            value=""
          />
        </div>
        <p
          className="ts-psa__project-form__finance__subtitle"
        >
          Target Revenue
        </p>
        <div
          className="ts-psa__form-field  "
        >
          <div
            className="ts-psa__form-field__label"
          >
             
            Planned Revenue
          </div>
          <input
            className="ts-text-field slds-input"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder={null}
            readOnly={false}
            type="text"
            value=""
          />
        </div>
        <div
          className="ts-psa__form-field  "
        >
          <div
            className="ts-psa__form-field__label"
          >
             
            Margin
          </div>
          <input
            className="ts-text-field slds-input"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder={null}
            readOnly={false}
            type="text"
            value=""
          />
        </div>
        <div
          className="ts-psa__form-field  "
        >
          <div
            className="ts-psa__form-field__label"
          >
             
            Planned Cost
          </div>
          <input
            className="ts-text-field slds-input"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder={null}
            readOnly={false}
            type="text"
            value=""
          />
        </div>
        <div
          className="ts-psa__form-field  "
        >
          <div
            className="ts-psa__form-field__label"
          >
             
            Man-Hour Revenue
          </div>
          <input
            className="ts-text-field slds-input"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder={null}
            readOnly={false}
            type="text"
            value=""
          />
        </div>
        <div
          className="ts-psa__form-field  "
        >
          <div
            className="ts-psa__form-field__label"
          >
             
            Man-Hour Cost
          </div>
          <input
            className="ts-text-field slds-input"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder={null}
            readOnly={false}
            type="text"
            value=""
          />
        </div>
      </section>
    </div>
  </div>
</div>
`;
